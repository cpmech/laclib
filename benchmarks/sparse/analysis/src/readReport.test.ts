import { readReport } from './readReport';

describe('readReport', () => {
  it('reads bfwb62', () => {
    const rpt = readReport('bfwb62');
    expect(rpt).toEqual({
      Platform: 'laclib',
      SolverKind: 'mumps',
      MatrixName: 'bfwb62',
      Ordering: 'metis',
      MpiSize: 1,
      Symmetric: true,
      NumberOfRows: 62,
      NumberOfCols: 62,
      NumberOfNonZeros: 202,
      StepReadMatrix: {
        ElapsedTimeNanoseconds: 300320,
        ElapsedTimeString: '300.32µs',
        MemoryBytes: 333336576,
        MemoryMiB: '317',
      },
      StepAnalyze: {
        ElapsedTimeNanoseconds: 1808093,
        ElapsedTimeString: '1.80809ms',
        MemoryBytes: 333516800,
        MemoryMiB: '318',
      },
      StepFactorize: {
        ElapsedTimeNanoseconds: 1219917,
        ElapsedTimeString: '1.21992ms',
        MemoryBytes: 15366782976,
        MemoryMiB: '14,654',
      },
      StepSolve: {
        ElapsedTimeNanoseconds: 792976,
        ElapsedTimeString: '792.976µs',
        MemoryBytes: 15366782976,
        MemoryMiB: '14,654',
      },
      TimeSolverNanoseconds: 3313270,
      TimeSolverString: '3.31327ms',
      Stats: {
        NormInfA: 0.0001,
        NormInfAx: 1,
        NormInfDiff: 6.66133814775094e-16,
        RelativeError: 6.66067208054288e-16,
        ElapsedTimeNanoseconds: 6440,
        ElapsedTimeString: '6.44µs',
      },
    });
  });

  it('fails due to wrong matrixName', () => {
    expect(() => {
      readReport('nada');
    }).toThrowError(
      "ENOENT: no such file or directory, open '../results/mumps_nada_metis_np1.json'",
    );
  });
});
