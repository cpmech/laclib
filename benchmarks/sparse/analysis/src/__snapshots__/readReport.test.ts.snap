// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readReport reads bfwb62 1`] = `
Object {
  "MatrixName": "bfwb62",
  "MpiSize": 1,
  "NumberOfCols": 62,
  "NumberOfNonZeros": 202,
  "NumberOfRows": 62,
  "OmpNumThreads": 0,
  "Ordering": "metis",
  "Platform": "laclib",
  "SolverKind": "mumps",
  "Stats": Object {
    "ElapsedTimeNanoseconds": 5013,
    "ElapsedTimeString": "5.013µs",
    "NormInfA": 0.0001,
    "NormInfAx": 1,
    "NormInfDiff": 6.66133814775094e-16,
    "RelativeError": 6.66067208054288e-16,
  },
  "StepAnalyze": Object {
    "ElapsedTimeNanoseconds": 1657060,
    "ElapsedTimeString": "1.65706ms",
    "MemoryBytes": 333520896,
    "MemoryMiB": "318",
  },
  "StepFactorize": Object {
    "ElapsedTimeNanoseconds": 1239142,
    "ElapsedTimeString": "1.23914ms",
    "MemoryBytes": 15366787072,
    "MemoryMiB": "14,654",
  },
  "StepReadMatrix": Object {
    "ElapsedTimeNanoseconds": 293547,
    "ElapsedTimeString": "293.547µs",
    "MemoryBytes": 333340672,
    "MemoryMiB": "317",
  },
  "StepSolve": Object {
    "ElapsedTimeNanoseconds": 737954,
    "ElapsedTimeString": "737.954µs",
    "MemoryBytes": 15366787072,
    "MemoryMiB": "14,654",
  },
  "Symmetric": true,
  "TimeSolverNanoseconds": 3136812,
  "TimeSolverString": "3.13681ms",
}
`;
